GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.cs

```diff
@@ -133,118 +137,87 @@ protected override void PageToSettings()
             }
         }
 
-        private void GlobalMergeTool_TextChanged(object sender, EventArgs e)
+        private string BrowseDiffMergeTool(string toolName, string path)
         {
-            if (IsLoadingSettings)
+            var diffMergeToolConfig = _diffMergeToolConfigurationManager.LoadDiffMergeToolConfig(toolName, null);
+
+            var initialDirectory = Path.GetDirectoryName(path);
+            if (string.IsNullOrWhiteSpace(initialDirectory))
             {
-                return;
+                initialDirectory = Path.GetDirectoryName(diffMergeToolConfig.Path);
             }
 
-            var mergeTool = _NO_TRANSLATE_GlobalMergeTool.Text.Trim();
-            MergetoolPath.Text = CurrentSettings.GetValue($"mergetool.{mergeTool}.path");
-            MergeToolCmd.Text = CurrentSettings.GetValue($"mergetool.{mergeTool}.cmd");
+            var filter = !string.IsNullOrWhiteSpace(diffMergeToolConfig.ExeFileName)
+                ? string.Format("{0}|{1}", toolName, diffMergeToolConfig.ExeFileName)
+                : string.Format("{0} (*.exe;*.cmd;*.bat)|*.exe;*.cmd;*.bat", toolName);
```

Comment:
As the toolname is set to 'toolname.exe' for unknown tools, it can be hard to find the exe
Should probably be changed in 
        public DiffMergeToolConfiguration LoadDiffMergeToolConfig(string toolName, string userSuppliedPath)

Reply:
Could you please clarify?

Reply:
If I add my tool specificTool, the filter is set to "specificTool.exe" which may not be what I expect

Reply:
Gotcha.

Reply:
fixed in 68ce7dcfbf88c762ef0ff47d63be42be59241bd0